### PR TITLE
Wsteam1 164 storybook toolbar

### PR DESCRIPTION
**Pull request by DarioR01**
Resolves [WSTEAM1-290](https://jira.dev.bbc.co.uk/browse/WSTEAM1-290)

**Overall change:**
Creates storybook Sidebar labels visualising health factors summary

**Code changes:**

- Created a new storybook addon called './SidebarLabel/preset.cjs'
- Created a new SidebarLabel component 
- Read component documentation for better understanding

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: .storybook/DocsDecorator/HealthFactors/SingleDoc/index.styles.ts [hunk 1]
```diff
@@ -12,13 +12,12 @@ const styles = {
...
-    css({
-      width: `${pixelsToRem(14)}rem`,
-      height: `${pixelsToRem(14)}rem`,
-      marginLeft: '0.5em',
-      verticalAlign: `-${pixelsToRem(2)}rem`,
-    }),
+  linkIconContainer: {
+    width: `${pixelsToRem(14)}rem`,
+    height: `${pixelsToRem(14)}rem`,
+    marginLeft: '0.5em',
```
Comment (greenc05) [inline .storybook/DocsDecorator/HealthFactors/SingleDoc/index.styles.ts]: Should this be 'em'?

File: .storybook/DocsDecorator/HealthFactors/index.styles.tsx [hunk 2]
```diff
@@ -30,13 +30,12 @@ const styles = {
...
-    css({
-      width: `${pixelsToRem(24)}rem`,
-      height: `${pixelsToRem(24)}rem`,
-      marginRight: '0.5em',
-      display: 'block',
-    }),
+  titleIcon: {
+    width: `${pixelsToRem(24)}rem`,
+    height: `${pixelsToRem(24)}rem`,
+    marginRight: '0.5em',
```
Comment (greenc05) [inline .storybook/DocsDecorator/HealthFactors/index.styles.tsx]: em?

File: .storybook/SidebarLabel/index.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import pixelsToRem from '../../src/app/utilities/pixelsToRem';
+
+const styles = {
+  iconWrapper: {
+    display: 'inline-block',
+    boxSizing: 'border-box',
+    marginLeft: `0.5em`,
```
Comment (greenc05) [inline .storybook/SidebarLabel/index.styles.tsx]: em?

File: .storybook/SidebarLabel/index.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  labelWrapper: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+
+  warningIcon: {
+    width: `${pixelsToRem(12)}rem`,
+    height: `${pixelsToRem(12)}rem`,
+    fill: '#3f3f42',
```
Comment (greenc05) [inline .storybook/SidebarLabel/index.styles.tsx]: Please could we update this colour to #c64f00 (assuming this is for the outstanding icon - triangle with exclamation mark inside)

**Comment by greenc05** [comment]
Are we missing the goodness to force the icons to be the user's colours if they change colour preferences in Firefox etc?

File: .storybook/SidebarLabel/index.styles.tsx (hunk 4 above)
Comment (DarioR01) [reply .storybook/SidebarLabel/index.styles.tsx]: Updated

File: .storybook/DocsDecorator/HealthFactors/SingleDoc/index.styles.ts (hunk 1 above)
Comment (DarioR01) [reply .storybook/DocsDecorator/HealthFactors/SingleDoc/index.styles.ts]: Webcore uses em, I am assuming it was a UX requirement on their side to have the spacing depending on font size.

File: .storybook/DocsDecorator/HealthFactors/index.styles.tsx (hunk 2 above)
Comment (DarioR01) [reply .storybook/DocsDecorator/HealthFactors/index.styles.tsx]: Webcore uses em, I am assuming it was a UX requirement on their side to have the spacing depending on font size.

File: .storybook/SidebarLabel/index.styles.tsx (hunk 3 above)
Comment (DarioR01) [reply .storybook/SidebarLabel/index.styles.tsx]: Webcore uses em, I am assuming it was a UX requirement on their side to have the spacing depending on font size.

**Comment by DarioR01** [comment]
> Are we missing the goodness to force the icons to be the user's colours if they change colour preferences in Firefox etc?

This is now fixed 😓. Just a case of adding color and current color styles. see:  [fix: a11y custom icon colours](https://github.com/bbc/simorgh/pull/10484/commits/ec08e23e798b4aafd97d8d6bb9696d5a02e93f4f)

**Comment by pvaliani** [comment]
Just reviewing this with @louisearchibald. Currently most components show the data missing icon because component health missing. Is there an automatic way we are going to retrofit component health for current existing components to get them where they need to be? i.e green

**Comment by DarioR01** [comment]
> Just reviewing this with @louisearchibald. Currently most components show the data missing icon because component health missing. Is there an automatic way we are going to retrofit component health for current existing components to get them where they need to be? i.e green

No, It is going to be a manual work, and the idea of this PR is to help with this manual work by giving a better visualisation of what is missing, what is outstanding, and what we have already completed. 
The best I can do is to automate the creation of outstanding actions (i.e. yellow), as an empty file will suffice. But to complete the health factors we must find and fill out the missing documentation.

**Comment by greenc05** [comment]
@pvaliani @DarioR01 @louisearchibald  Yeah we aren't going to retrofit component health but I would be happy to help find what docs I can if someone wants to add what we can find on a learning day or something 🤔

**Comment by greenc05** [comment]
@DarioR01 minor comments sent to you, then lets a11y swarm this 🙌

**Review by greenc05 — APPROVED** [review]
@DarioR01 minor comments sent to you, then lets a11y swarm this 🙌

**Review by pvaliani — APPROVED** [review]

**Pull request merged**